cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceClassLoader.java

```diff
@@ -140,6 +142,10 @@ public MapReduceTaskContextProvider getTaskContextProvider() {
     LoggingContext loggingContext = createMapReduceLoggingContext();
     LoggingContextAccessor.setLoggingContext(loggingContext);
 
+    // Redirect stdout and stderr for mapreduce to logger
+    System.setOut(new PrintStream(RedirectedPrintStream.createRedirectedOutStream(LOG, System.out), true));
```

Comment:
Instead of doing it in getTaskContextProvider(), we can create a new method mapReduceLogRedirector() and call it from MapReduceContainerLauncher. What do you think @chtyim